gradle-incremental-configuration-cache/src/main/java/com/palantir/gradle/configcache/incremental/DryRunConfigurationCacheEnabledTask.java

```diff
@@ -0,0 +1,210 @@
+            return;
+        }
+
+        String message = errorMessage(Optional.ofNullable(((Failure) result).errorOutput()));
+        throw new RuntimeException(message);
+    }
+
+    private String errorMessage(Optional<String> output) {
+        boolean isCircleCI = getEnvironmentVariables()
+                .envVarOrFromTestingProperty("CIRCLECI")
+                .isPresent();
+
+        if (!isCircleCI) {
+            return buildDetailedErrorMessage(output, Optional.empty());
+        }
+
+        return circleCiErrorMessage(output);
+    }
+
+    private String circleCiErrorMessage(Optional<String> output) {
+        try {
+            String artifactPath = "configuration-cache-validation-report/validation-report.txt";
+            ArtifactLocation artifactLocation =
+                    getCircleCiArtifacts().resolveArtifactLocation(artifactPath).get();
+
+            Path artifactLocationPath =
+                    artifactLocation.physicalPath().getAsFile().toPath();
+            Files.createDirectories(artifactLocationPath.getParent());
+            Files.writeString(artifactLocationPath, output.orElse(""));
+
+            return buildDetailedErrorMessage(output, Optional.of(artifactLocation.circleLink()));
+        } catch (IOException e) {
+            // Fall back to including output directly if we can't create the artifact
+            return String.format(
+                    """
+                    %s
+
+                    (Failed to save CircleCI artifact: %s)
+                    """,
+                    buildDetailedErrorMessage(output, Optional.empty()), e.getMessage());
+        }
+    }
+
+    @SuppressWarnings("checkstyle:LineLength")
+    private String buildDetailedErrorMessage(Optional<String> outputContent, Optional<String> validationReportUrl) {
+        Optional<String> configCacheReportPath = extractConfigCacheReportPath(outputContent);
+
+        // Get the proper artifact URL for the configuration cache report
+        Optional<String> configCacheReportUrl = Optional.empty();
+        if (configCacheReportPath.isPresent() && validationReportUrl.isPresent()) {
+            configCacheReportUrl = configCacheReportArtifactUrl(configCacheReportPath.get());
+        }
+
+        // Build the reports section based on what's available
+        String reportsSection = reportsSection(validationReportUrl, configCacheReportUrl, configCacheReportPath);
+
+        // Build the main message (shared between CI and local)
+        String mainMessage = String.format(
+                """
+            ❌ CONFIGURATION CACHE ALLOW LIST VALIDATION FAILED
+
+            WHAT HAPPENED:
+              Some task / tasks in the allow list failed to run with configuration cache enabled.
+
+            %s
+
+            WHY THIS MATTERS:
+              This validation task runs all the tasks marked as configuration cacheable in the allow list,
+              to ensure the allow list contains only cacheable tasks.
```

Comment:
we should be a bit more specific with language here.

It's not that the tasks are cacheable, but they are configuration cache compatible.

Cacheable tasks are their own separate thing